client/python/datajunction/client.py

```diff
@@ -547,16 +592,16 @@ def sql(  # pylint: disable=too-many-arguments
         return response.json()
 
     @staticmethod
-    def results_to_dataframe(results) -> pd.DataFrame:
+    def process_results(results) -> Results:
```

Comment:
@shangyian I changed this to just return the data and want to get your thoughts on it. The two main benefits I see here is that the datajunction client would no longer have to include pandas as a dependency and we can avoid an assumption that the user's end goal is a pandas dataframe. Here are some examples of one-liners that can get from this `Results` data model to a number of other in-memory data formats including pandas.

_directly to a pandas dataframe_
```py
import pandas as pd
df = pd.DataFrame(**results)
```
_directly to a pyarrow table_
```py
import pyarrow as pa
table = pa.Table.from_arrays(list(map(list, zip(*results['data']))), names=results['columns'])
```
_directly to a partitioned dask dataframe_
```py
from dask import dataframe as dd
ddf = dd.from_dict({k:v for k, v in zip(results['columns'], list(map(list, zip(*results['data']))))}, npartitions=4)
```
_directly to a distributed modin dataframe (using their drop-in replacement for pandas)_
```py
import modin.pandas as pd
mdf = pd.DataFrame(**results)
```

Reply:
I think my main concern here is that we should try to give users exactly what they want, whether that's pandas, pyarrow or something else, rather than our `Results` object. We can certainly have that as a part of an intermediary stage on the client, but we should also include the conversion functions you have above and provide all of them to the user, and then have several versions of the client like `datajunction[pandas]`, `datajunction[pyarrow]`, each of which would include the dependencies to support those specific libraries.

Reply:
That's a fair compromise, I'll update this with your proposal. 👍